clang/include/clang/AST/Expr.h

```diff
@@ -1752,7 +1752,8 @@ enum class StringLiteralKind {
   UTF8,
   UTF16,
   UTF32,
-  Unevaluated
+  Unevaluated,
+  Binary
```

Comment:
Can you add a comment explaining this is for embed?

I'm sorry it took me a while to understand how this patch works.
(The reason is that this allows us to not "cast" to char in `getCodeUnitS()` - which is only used in C23 mode)

Maybe also add a comment in `getCodeUnitS` and/or `CheckC23ConstexprInitStringLiteral`

Reply:
Yup, Thanks!